src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,688 @@
+                : generator.IdentifierName(methodSymbol.Name);
+
+            var invocation = generator.InvocationExpression(memberName, arguments);
+            List<SyntaxNode> invocationReturn;
+
+            if (methodSymbol.ReturnsVoid)
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    invocation
+                };
+            }
+            else
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    generator.ReturnStatement(invocation)
+                };
+            }
+
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol);
+            var newMethodDeclaration = generator.MethodDeclaration(newMethod, invocationReturn);
+            editor.InsertBefore(oldMethodDeclaration, newMethodDeclaration);
+        }
+
+        public static void UpdateExpressionInOriginalFunction(SemanticDocument semanticDocument,
+            TExpressionSyntax expression, SyntaxNode scope, string parameterName, SyntaxEditor editor,
+            bool allOccurrences, CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(semanticDocument.Document);
+            var matches = FindMatches(semanticDocument, expression, semanticDocument, scope, allOccurrences, cancellationToken);
+            var parameterNameSyntax = (TIdentifierNameSyntax)generator.IdentifierName(parameterName);
+
+            // Parenthesize the variable, and go and replace anything we find with it.
+            // NOTE: we do not want elastic trivia as we want to just replace the existing code 
+            // as is, while preserving the trivia there.
+            var replacement = generator.AddParentheses(parameterNameSyntax, includeElasticTrivia: false);
+
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            foreach (var match in matches)
+            {
+                if (match.Equals(expression))
+                {
+                    if (ShouldRemoveVariableDeclaratorContainingExpression(semanticDocument, expression, out var varDecl))
+                    {
+                        var localDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => syntaxFacts.IsLocalDeclarationStatement(node));
+                        if (localDeclaration is not null)
+                        {
+                            editor.RemoveNode(localDeclaration);
+                        }
+                    }
+                    else
+                    {
+                        parameterNameSyntax = (TIdentifierNameSyntax)generator.IdentifierName(generator.Identifier(parameterName).WithAdditionalAnnotations(RenameAnnotation.Create()));
+                        replacement = generator.AddParentheses(parameterNameSyntax, includeElasticTrivia: false);
+                        editor.ReplaceNode(match, replacement);
```

Comment:
i honestly don't get why we're adding parens for the replacements.  is there a case where it's necessary to insert `(p)` instead of just `p`?  If so, doc it in the code.

Reply:
I don't think so, it ended up just getting simplified and removed anways